src/App.tsx

```diff
@@ -1,25 +1,68 @@
 import React from 'react';
-import { useSelector } from 'react-redux';
+import { useSelector, useDispatch } from 'react-redux';
 
 import './App.scss';
-import Start from './components/Start';
-import { Finish } from './components/Finish';
-
-import { isLoading, getMessage } from './store';
+import { getTodos, getUsers } from './helpers/api';
 
+import {
+  startLoading,
+  finishLoading,
+  handleSuccess,
+  handleError,
+  hasError,
+  isLoading,
+  listOfTodos,
+} from './store';
+import TodoList from './components/TodoList';
 
 const App = () => {
+  const dispatch = useDispatch();
   const loading = useSelector(isLoading);
-  const message = useSelector(getMessage) || 'Ready!';
+  const error = useSelector(hasError);
+  const todos = useSelector(listOfTodos);
 
-  return (
-    <div className="App">
-      <h1>Redux list of todos</h1>
-      <h2>{loading ? 'Loading...' : message}</h2>
+  const getData = () => {
+    dispatch(startLoading());
+
+    Promise.all([getTodos(), getUsers()])
+      .then(([todoFromServer, usersFromServer]) => {
+        const preparedTodosList = todoFromServer.map(post => {
+          return {
+            ...post,
+            user: usersFromServer
+              .find(user => user.id === post.userId),
+          };
+        });
 
-      <Start title="Start loading" />
-      <Finish title="Succeed loading" message="Loaded successfully!" />
-      <Finish title="Fail loading" message="An error occurred when loading data." />
+        dispatch(handleSuccess(preparedTodosList as Todo[]));
+      })
+      .catch(() => {
+        dispatch(handleError());
```

Comment:
```suggestion
      .catch((e) => {
        dispatch(handleError(e.message || e));
```
Было бы не плохо передавать саму ошибку в стор и выводить ее как-то на экран, чтобы в случае чего пользователь знал хотя бы что пошло не так

Reply:
Добавил вывод сообщения ошибки.